Conformance additions for 2.2 (#1426)
finos/FDC3: website/versioned_docs/version-2.0/api/conformance/Intents-Tests.md

-  `IntentAppDMultiple3`: Calls `fdc3.findIntent("sharedTestingIntent1", 'testContextY')`.  Receives promise containing an appIntent with metadata containing `sharedTestingIntent` and only **B** app metadata.

### Find Intents By Context

-  `SingleContext`: Call `fdc3.findIntentsByContext("testContextX")`.  Should return `aTestingIntent` (app **A**), `sharedTestingIntent1` (**A**, **B**) and `cTestingIntent` (**C**) AND nothing else.
-  `NoContext`: Call `fdc3.findIntentsByContext()`. Throws error of `NoAppsFound`

### Raise Intent

| App | Step           | Details                                                                                           |
|-----|----------------|---------------------------------------------------------------------------------------------------|
| D   | 1. Raise          | `fdc3.raiseIntent("sharedTestingIntent1", {testContextY})`<br />starts app B.                       |
| B   | 2. Gather Context | `fdc.addIntentListener("sharedTestingIntent1")`<br />Receives testContextY, matching that sent by D |

-  `SingleResolve1`: Perform above test

> -  `TargetedResolve1`: Use `fdc3.raiseIntent("aTestingIntent", {testContextX}, <A"s App Name>)` to start app A, otherwise, as above
> -  `TargetedResolve2`: Use `fdc3.raiseIntent("aTestingIntent", {testContextX}, {name: "<A's App Name>"})` to start app A, otherwise, as above
> -  `TargetedResolve3`: Use `fdc3.raiseIntent("aTestingIntent", {testContextX}, {name: "<app B Name>", appId: "<app B ID>"})` to start app B, otherwise, as above
> -  `FailedResolve1-3` As with `TargetedResolve1-3`, but use `fdc3.raiseIntent("aTestingIntent", {testContextY}, <A"s App Name>)` and variations.  You will receive `NoAppsFound` Error
> -  `FailedResolve4` As above, but use `fdc3.raiseIntent("aTestingIntent", {testContextX}, <C"s App Name>)`.  You will receive `NoAppsFound` Error

Reviewer: I think there are some places where `'` is still needed (and some inconsistency):
```suggestion
-  `TargetedResolve1`: Use `fdc3.raiseIntent("aTestingIntent", {testContextX}, <A's App Name>)` to start app A, otherwise, as above
-  `TargetedResolve2`: Use `fdc3.raiseIntent("aTestingIntent", {testContextX}, {name: "<A's App Name>"})` to start app A, otherwise, as above
-  `TargetedResolve3`: Use `fdc3.raiseIntent("aTestingIntent", {testContextX}, {name: "<B's App Name>", appId: "<B's App ID>"})` to start app B, otherwise, as above
-  `FailedResolve1-3` As with `TargetedResolve1-3`, but use `fdc3.raiseIntent("aTestingIntent", {testContextY}, <A's App Name>)` and variations.  You will receive `NoAppsFound` Error
-  `FailedResolve4` As above, but use `fdc3.raiseIntent("aTestingIntent", {testContextX}, <C's App Name>)`.  You will receive `NoAppsFound` Error
```

- `2.0-RaiseIntentSingleResolve`: Perform above test
- `2.0-RaiseIntentTargetedAppResolve`: Repeat the above test, but:

>   - In the first step use `fdc3.raiseIntent("sharedTestingIntent1", testContextX, {"appID": "<B"s appId>"})` to start app B,

Reviewer: ```suggestion
  - In the first step use `fdc3.raiseIntent("sharedTestingIntent1", testContextX, {"appID": "<B's appId>"})` to start app B,
```

Author: @robmoffat and this one.

| App  | Step                 | Details                                                                                                                                    |
|-------|-----------------|---------------------------------------------------------------------------------------------------|

> | Test   | 1. Raise intent | Test raises an intent with `fdc3.raiseIntent(""kTestingIntent", testContextX, {appId: "<K's appId>"})`<br />starts app K. |

Reviewer: ```suggestion
| Test   | 1. Raise intent | Test raises an intent with `fdc3.raiseIntent("kTestingIntent", testContextX, {appId: "<K's appId>"})`<br />starts app K. |
```

Author: @robmoffat and this one
